src/components/Modal.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect } from "react"
+import ModalCmp from "react-modal"
+import { DONT_SHOW_PRIVATE_GITHUB_WARNING_KEY } from "../theme/Root"
+import styles from "./Modal.module.css"
+
+interface ModalProps {
+  externalLink: string
+  showModal: boolean
+  children?: React.ReactNode
+  shouldCloseOnEsc?: boolean
+  shouldAcceptOnEnter?: boolean
+  shouldCloseOnOverlayClick?: boolean
+  handleCancelRequest: () => void
+  handleAcceptRequest?: () => void
+}
+
+export const idOfNoticeLink = "notice"
+
+if (typeof window !== "undefined") {
+  ModalCmp.setAppElement("body")
+}
+
+export const SubscriptionNoticeModal: React.FC<ModalProps> = ({
+  externalLink,
+  showModal,
+  shouldCloseOnEsc = true,
+  shouldAcceptOnEnter = false,
+  shouldCloseOnOverlayClick = true,
+  handleCancelRequest,
+  handleAcceptRequest,
+}) => {
+  const onRequestClose = () => {
+    // If the user checked to never see this notice but subsequently cancels we will disregard their selection. We will
+    // only stop showing this notice if they check the box and then proceed to GitHub
```

Comment:
I see that this is intentional - Is there a particular reason for disregarding it?

Reply:
We can discuss this quickly next time we speak - but for the purpose of this PR we _do_ want to have that behavior. It's a slightly better user experience as it applies your wishes only after you've chosen to follow the link forward. Otherwise you can get into a s situation where you dind't mean to dismiss the thing and never see it again.

Reply:
👍🏾